### PR TITLE
Fix vote by email

### PR DESCRIPTION
**Pull request by HSalmon**
Fix for: https://github.com/loomio/loomio/issues/3594

I introduced this bug when I tried to fix issue https://github.com/loomio/loomio/issues/3169 with this PR: https://github.com/loomio/loomio/pull/3273

### CODE REVIEW

File: angular/core/components/thread_page/thread_page_controller.coffee [hunk 1]
```diff
@@ -3,7 +3,7 @@ angular.module('loomioApp').controller 'ThreadPageController', ($scope, $routePa
 
   @requestedProposalKey = $routeParams.proposal or $location.search().proposal
   @requestedCommentId   = parseInt($routeParams.comment or $location.search().comment)
-  $location.url($location.path())
+  $location.url($location.path()) unless $location.search().position
```
Comment (robguthrie) [inline angular/core/components/thread_page/thread_page_controller.coffee]: Can you explain the behaviour here, @HSalmon?
When will we see a search().position be present, what is the user doing in that case?

Comment (HSalmon) [inline angular/core/components/thread_page/thread_page_controller.coffee]: Voting via email

Comment (HSalmon) [inline angular/core/components/thread_page/thread_page_controller.coffee]: Actually, it occurred to me that there are probably other times when we don't want to clear the url e.g. when a person is following a link in a membership request email (i.e. someone has requested access to one of their groups).

**Comment by gdpelican** [comment]
Pretty sure we want to clear the route params, but we want to do it later, (ie, after we've called `openVoteModal`)

How about doing it at the end of the, possibly now ill-named, `performScroll` method?

**Pull request merged**